src/client/app/components/RouteComponent.tsx

```diff
@@ -54,31 +54,36 @@ export default class RouteComponent extends React.Component<RouteProps> {
 		this.requireRole = this.requireRole.bind(this);
 	}
 
+	/**
+	 * TODO The following three functions, requireRole, requireAuth, and checkAuth, do not work exactly as intended.
+	 * Their async blocks evaluate properly, but the returns inside of them are never honored. The end return statement is always what is evaluated.
+	 * Fixing this may require some major changes to how page redirects are done. This is detailed more in issue #817.
+	 * The errors can be obtained by putting breakpoints on all returns and then stepping through a page load in a debugger.
+	*/
+
 	/**
 	 * Generates middleware that requires proper role and authentication for a page route
 	 * @param requiredRole The role that is necessary to access a page route
 	 * @param component The component of the page redirecting
-	 * @returns middleware that requires proper role and authentication for a page route
+	 * @returns The page route to continue to (component or home)
 	 */
 	public requireRole(requiredRole: UserRole, component: JSX.Element) {
-		// Redirect route to login page if the auth token does not exist
-		if (!hasToken()) {
-			return <Redirect to='/login'/>;
+		// Redirect route to login page if the auth token does not exist or if the user is not a csv role
```

Comment:
I'm unclear on why this comment has csv role since the code seems to be using value passed? Am I missing something?

Reply:
That would be a typo on my part, since the only use of that function right now is for the csv page.